Introduces history manager screen for medical manager

mtabishk opened the pull request:
https://user-images.githubusercontent.com/61789893/193395332-ccc84651-144f-4287-b319-02c291c11246.mp4



1. Added History Manager UI
2. API Integration with backend

ashutoshc8101 commented on api/medical/models.py:
```diff
@@ -55,9 +55,10 @@ class Appointment(models.Model):
-    start_time = models.TimeField(null = True)
-    end_time = models.TimeField(null = True)
-    status = models.CharField(choices=APPOINTMENT_STATUS, default="IN PROGRESS", max_length=30)
+    start_time = models.TimeField(null=True)
+    end_time = models.TimeField(null=True)
+    status = models.CharField(
+        choices=APPOINTMENT_STATUS, default="IN PROGRESS", max_length=30)
```
Changes not required

mtabishk replied:
Done

ashutoshc8101 commented on api/medical/models.py:
```diff
@@ -69,8 +70,6 @@ class MedicalHistory(models.Model):
         settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
     doctor = models.ForeignKey(Doctor, on_delete=models.CASCADE)
     date = models.DateField(null=True)
-    time = models.TimeField(null=True)
```
Don't remove the time field.

mtabishk replied:
Done

ashutoshc8101 commented on api/medical/views.py:
```diff
@@ -233,13 +233,14 @@ def post(self, request):
+            name=name,
+            designation=profession,
+            phone=contact
+        )
+
+        staff.save()
+        return Response(status=200, data={'msg': 'Staff added successfully'})
 
-        staff.save();
-        return Response(status = 200, data={ 'msg': 'Staff added successfully' })
```
Changes not required.

mtabishk replied:
Done

ashutoshc8101 commented on api/medical/views.py:
```diff
@@ -306,26 +307,28 @@ def post(self, request):
         now = datetime.datetime.now()
 
-        appointments = Appointment.objects.filter(status = "ACCEPTED", date__gt=now.date()).all()
+        appointments = Appointment.objects.filter(
+            status="ACCEPTED", date__gt=now.date()).all()
```
don't push changes that only change the indentation.

mtabishk replied:
Done

ashutoshc8101 commented on app/env.symlink:
```diff
@@ -0,0 +1 @@
+../.env
```
Don't push changes to `env.symlink`

mtabishk replied:
Okay

ashutoshc8101 commented on app/lib/main_dev.dart:
```diff
@@ -21,6 +21,7 @@ Future main() async {
 
   await Firebase.initializeApp(
     options: DefaultFirebaseOptions.currentPlatform,
+    name: 'ira',
```
Why is this required?

mtabishk replied:
It was required on my side. There was some issue "Duplicate Firebase APP"

ashutoshc8101 replied:
Remove this field if this doesn't apply to all contributors.

mtabishk replied:
Removed

ashutoshc8101 commented on app/lib/screens/medical/manager/history/doctor_model.dart:
```diff
@@ -0,0 +1,38 @@
+  final String mail;
+  final String date;
+  // ignore: non_constant_identifier_names
+  final String start_time;
+  // ignore: non_constant_identifier_names
+  final String end_time;
```
Use `camelCase` here.

mtabishk replied:
Done

ashutoshc8101 commented on app/lib/screens/medical/manager/history/doctor_model.dart:
```diff
@@ -0,0 +1,38 @@
+class DoctorModel {
+  DoctorModel(
+    this.id,
+    this.name,
+    this.phone,
+    this.specialization,
+    this.mail,
+    this.date,
+    this.start_time,
+    this.end_time,
+    this.details,
+  );
```
```suggestion
  DoctorModel({
    required this.id,
    required this.name,
    required this.phone,
    required this.specialization,
    required this.mail,
    required this.date,
    required this.start_time,
    required this.end_time,
    required this.details,
    }
  );
```

mtabishk replied:
Why is this required?

ashutoshc8101 commented on app/lib/screens/medical/manager/history/history_manager.dart:
```diff
@@ -0,0 +1,168 @@
+
+class StudentModel {
+  // ignore: non_constant_identifier_names
+  final String first_name;
+  // ignore: non_constant_identifier_names
```
Use camelCase

mtabishk replied:
Done

ashutoshc8101 reviewed (changes requested):
Left few comments and dont push `app/flutter_04.png`

ashutoshc8101 commented:
@Jeetu303 Please review video.

Jeetu303 commented:
Seems good
Go ahead

ashutoshc8101 commented on app/.env.symlink:
```diff
@@ -1 +0,0 @@
-../.env
```
This file shouldn't be deleted.

mtabishk replied:
Okay. I have added .env.symlink from ira main branch

ashutoshc8101 replied:
This file should be a symlink. Ideally, changes to this file shouldn't be even here in diff.

ashutoshc8101 reviewed (changes requested):
Not all comments are properly addressed. Also, flutter analyze is still failing. Ask for review, once all comments are properly addressed.

mtabishk commented:
@ashutoshc8101 Flutter analyze has no issues on my end. I have added .env.symlink from ira main branch and addressed all comments
![Screenshot 2022-10-01 133640](https://user-images.githubusercontent.com/61789893/193399946-508a5ef0-62e6-4351-a370-985359c03533.png)
